### PR TITLE
Put things where they belong in a gem

### PR DESCRIPTION
**Pull request by KnVerey**
@kirs @sirupsen

This simply moves code around to split classes into their own files and move the files to their conventional locations for a gem. Let me know if something still isn't quite right!

The only code change is to make `KubernetesDeploy` a module and name the extracted runner class `KubernetesDeploy::Runner`, as Kir did on his branch.

### CODE REVIEW

File: lib/kubernetes-deploy/kubernetes_resource.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+%w(
+  configmap
+  deployment
+  ingress
+  persistent_volume_claim
+  pod
+  service
+).each do |subresource|
+  require "kubernetes-deploy/kubernetes_resource/#{subresource}"
```
Comment (kirs) [inline lib/kubernetes-deploy/kubernetes_resource.rb]: We usually put all `require`s in the top of the file. Can we do it here?

**Comment by kirs** [comment]
Thanks for working on it! This will be really helpful for testing the script.
I'm going to 🎩 this locally and report if it works.

**Comment by kirs** [comment]
I got a chance to run the branch locally and looks like something goes wrong under the hook (`Don't know how to monitor resources of type service`):

```
----------------------------------Phase 6: Deploying all resources----------------------------------
Deploying resources:
- service/redis
- persistentvolumeclaim/trashbin-redis
- deployment/redis
- pod/speaker-pod-6b300954-46d41383
- configmap/some-data
- ingress/web
- service/web
- deployment/web

[WARN]  Don't know how to monitor resources of type service. Assuming service/redis deployed successfully.
[KUBESTATUS] {"group":"services","name":"redis","status_string":"Unknown","exists":null,"succeeded":true,"failed":false,"timed_out"$false}
[WARN]  Don't know how to monitor resources of type persistentvolumeclaim. Assuming persistentvolumeclaim/trashbin-redis deployed s$ccessfully.
[KUBESTATUS] {"group":"persistentvolumeclaims","name":"trashbin-redis","status_string":"Unknown","exists":null,"succeeded":true,"fa$led":false,"timed_out":false}
[WARN]  Don't know how to monitor resources of type deployment. Assuming deployment/redis deployed successfully.
[KUBESTATUS] {"group":"deployments","name":"redis","status_string":"Unknown","exists":null,"succeeded":true,"failed":false,"timed_o$t":false}
[WARN]  Don't know how to monitor resources of type pod. Assuming pod/speaker-pod-6b300954-46d41383 deployed successfully.
[KUBESTATUS] {"group":"pods","name":"speaker-pod-6b300954-46d41383","status_string":"Unknown","exists":null,"succeeded":true,"faile$":false,"timed_out":false}
[WARN]  Don't know how to monitor resources of type configmap. Assuming configmap/some-data deployed successfully.
[KUBESTATUS] {"group":"configmaps","name":"some-data","status_string":"Unknown","exists":null,"succeeded":true,"failed":false,"time$_out":false}
[WARN]  Don't know how to monitor resources of type ingress. Assuming ingress/web deployed successfully.
[KUBESTATUS] {"group":"ingresss","name":"web","status_string":"Unknown","exists":null,"succeeded":true,"failed":false,"timed_out":f$lse}
[WARN]  Don't know how to monitor resources of type service. Assuming service/web deployed successfully.
[KUBESTATUS] {"group":"services","name":"web","status_string":"Unknown","exists":null,"succeeded":true,"failed":false,"timed_out":f$lse}
[WARN]  Don't know how to monitor resources of type deployment. Assuming deployment/web deployed successfully.
[KUBESTATUS] {"group":"deployments","name":"web","status_string":"Unknown","exists":null,"succeeded":true,"failed":false,"timed_out$:false}
Deploy succeeded!
```

**Comment by kirs** [comment]
```ruby
def self.for_type(type, name, namespace, file)
  if subclass = descendants.find { |subclass| subclass.handled_type.downcase == type }
    subclass.new(name, namespace, file)
  else
    self.new(name, namespace, file).tap { |r| r.type = type }
  end
end
```

It happens because calling `descendants` here will always return `[]`

**Comment by kirs** [comment]
Earlier, you had `class Service < KubernetesResource` and that allowed `DependencyTracker` to track it. In the new implementation, `Service` is not inherited from `KubernetesResource` anymore and this is the reason why it breaks.

**Comment by KnVerey** [comment]
Yes, that was dumb, dunno why I did that. It is fixed now. 🤦‍♀️

**Comment by KnVerey** [comment]
```
----------------------------------Phase 6: Deploying all resources----------------------------------
Deploying resources:
- Service/redis
- PersistentVolumeClaim/trashbin-redis
- Deployment/redis
- Pod/speaker-pod-d7cd30f2-397bfe5e
- Configmap/some-data
- Ingress/web
- Service/web
- Deployment/web
[KUBESTATUS] {"group":"Services","name":"redis","status_string":"1 endpoints","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"PersistentVolumeClaims","name":"trashbin-redis","status_string":"Bound","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Redis deployment","name":"redis-1471906532-97bzw","status_string":"Running (Ready: true)","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Deployments","name":"redis","status_string":"deployment \"redis\" successfully rolled out","exists":true,"succeeded":true,"failed":false,"timed_out":false,"replicas":{"updatedReplicas":1,"replicas":1,"availableReplicas":1},"num_pods":1}
[KUBESTATUS] {"group":"Bare pods","name":"speaker-pod-d7cd30f2-397bfe5e","status_string":"Succeeded (Ready: false)","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Configmaps","name":"some-data","status_string":"Available","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Ingresses","name":"web","status_string":"Created","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Services","name":"web","status_string":"1 endpoints","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Web deployment","name":"web-2583131178-1spkl","status_string":"Running (Ready: true)","exists":true,"succeeded":true,"failed":false,"timed_out":false}
[KUBESTATUS] {"group":"Deployments","name":"web","status_string":"deployment \"web\" successfully rolled out","exists":true,"succeeded":true,"failed":false,"timed_out":false,"replicas":{"updatedReplicas":1,"replicas":1,"availableReplicas":1},"num_pods":1}
Deploy succeeded!
```

**Comment by kirs** [comment]
👏 👏 👏

**Pull request merged**